### DISCUSSION TITLE
Forward doorbell button to get a notification on iPhone?

### DISCUSSION THREAD

**Discussion by likewise** [HomeSpan/HomeSpan#341] [General]
I have an RF wireless button at my door which is attached to a RF wireless doorbell (Byron DBY series).

I can receive and interpret the RF signal with my ESP32, so I have a doorbell button event. This works.

Is there a HomeKit service (and maybe iPhone app) that allows me to send a doorbell button event to the "Home" app, so that it rings my iPhone?

**Comment by HomeSpan** [comment]
Though Apple seems to have deprecated its Doorbell Service, you should be able to use Service::StatelessProgrammableSwitch within HomeSpan to trigger notifications on your iPhone.  See [Tutorial Example 16](https://github.com/HomeSpan/HomeSpan/blob/master/examples/16-ProgrammableSwitches) for details.

**Reply by likewise** [reply]
Thank you, this is what I tried yesterday, however for some (unknown for me) reason this requires a HomeKit Hub.

I think this might be due to the fact that it requires automation.

I find it strange this is all requires for just a doorbell button.

**Reply by likewise** [reply]
Of course, this is a HomeKit design issue, not related to HomeSpan, which seems a well designed library. Kudos!

**Reply by HomeSpan** [reply]
Good point - automation routines on an iPhone require the use of a HomeKit Hub (Apple TV, HomePod, etc.).  Not sure there is an easy alternative and agree it would make sense for Apple to include a standard mechanism.
